platforms/core-runtime/gradle-cli/src/main/java/org/gradle/launcher/cli/DefaultCommandLineActionFactory.java

```diff
@@ -166,25 +181,37 @@ private static class CommandLineParseFailureAction implements Action<ExecutionLi
         private final Exception exception;
         private final CommandLineParser parser;
 
-        public CommandLineParseFailureAction(CommandLineParser parser, Exception exception) {
+        private final List<String> args;
+
+        public CommandLineParseFailureAction(CommandLineParser parser, Exception exception, List<String> args) {
             this.parser = parser;
             this.exception = exception;
+            this.args = args;
         }
 
         @Override
         public void execute(ExecutionListener executionListener) {
             System.err.println();
             System.err.println(exception.getMessage());
-            showUsage(System.err, parser);
+            String suggestedTaskName = null;
+            for (String arg : args) {
+                if (!arg.startsWith("-")) {
```

Comment:
❔  Is there an existing string literal to match against to avoid using a magic constant?

Reply:
Nope:
* https://github.com/gradle/gradle/blob/a3dc91fa697b5de5fd9c2f4ce13e1fbda561fa26/platforms/core-runtime/cli/src/main/java/org/gradle/cli/CommandLineParser.java#L94